app/connectors/paritygateway/ParityGatewayClient.scala

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2016 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package connectors.paritygateway
+
+import config.MicroserviceAppConfig._
+import _root_.config.WSHttp
+import model.Exceptions.ConnectorException
+import play.api.http.Status._
+import play.api.libs.json.{ JsObject, JsValue }
+import uk.gov.hmrc.play.http.{ HeaderCarrier, HttpResponse }
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+object ParityGatewayClient extends ParityGatewayClient {
+  val http: WSHttp = WSHttp
+  val url = parityGatewayConfig.url
+}
+
+trait ParityGatewayClient {
+  val http: WSHttp
+  val url: String
+
+  implicit def blankedHeaderCarrier = new HeaderCarrier()
+
+  lazy val urlWithPathPrefix = s"$url/fset-parity-gateway/faststream"
```

Comment:
I think it does actually, i think I tested alternatives when working on the launchpad gateway. I think parityGatewayConfig is a lazy val, so even though the object is constructed at build time it doesn't actually resolve parityGatewayConfig until this call to url. If you try to load the config object too soon i don't think Play has properly initialised and it doesn't read correctly.